vshard/replicaset.lua

```diff
@@ -1091,17 +1091,15 @@ local function replicaset_locate_master(replicaset)
     local futures = {}
     local timeout = const_timeout
     local deadline = fiber_clock() + timeout
-    local async_opts = {is_async = true}
+    local async_opts = {is_async = true, timeout = timeout}
     local replicaset_id = replicaset.id
     for replica_id, replica in pairs(replicaset.replicas) do
-        local conn = replicaset_connect_to_replica(replicaset, replica)
-        if conn:is_connected() then
-            ok, f = pcall(conn.call, conn, func, args, async_opts)
-            if not ok then
-                last_err = lerror.make(f)
-            else
-                futures[replica_id] = f
-            end
+        replicaset_connect_to_replica(replicaset, replica)
+        ok, res, err = replica_call(replica, func, args, async_opts)
+        if not ok then
+            last_err = err
+        else
+            futures[replica_id] = res
```

Comment:
Please, do not send a request if the connection is not established. This function intentionally used to skip non-connected instances, hoping that at least among the connected ones a master would be found. You can still check it with `replica:is_connected()` to take into account the vconnect thing.

Reply:
Thank you for noticing, fixed!

Note, that I changed `replica:is_connected()` function, as otherwise it doesn't return error, only true or nil:

<details><summary>Current behavior of return values</summary>

```lua
tarantool> function a() return nil, 'error' end
---
...

tarantool> return true and true and a()
---
- null
...

tarantool> return true and true and nil, 'error'
---
- null
- error
...

```

</details>